XenAdmin/Wizards/ImportWizard/ImportWizard.cs

```diff
@@ -251,17 +251,30 @@ protected override void UpdateWizardContent(XenTabPage page)
                     m_pageImportSource.DiskCapacity, m_pageImportSource.IsWIM, m_pageVMconfig.AdditionalSpace,
                     m_pageImportSource.FilePath, m_pageImportSource.ImageLength);
 
+				if (m_pageStorage.SelectedOvfEnvelope != m_envelopeFromVhd
+					|| lunPerVdiMappingPage.SelectedOvfEnvelope != m_envelopeFromVhd
+					|| m_pageNetwork.SelectedOvfEnvelope != m_envelopeFromVhd)
+				{
+					ResetVmMappings(m_envelopeFromVhd);
+					NotifyNextPagesOfChange(m_pageHost, m_pageStorage, m_pageNetwork);
+				}
+
 				m_pageStorage.SelectedOvfEnvelope = m_envelopeFromVhd;
 			    lunPerVdiMappingPage.SelectedOvfEnvelope = m_envelopeFromVhd;
 				m_pageNetwork.SelectedOvfEnvelope = m_envelopeFromVhd;
-				ResetVmMappings(m_envelopeFromVhd);
-				NotifyNextPagesOfChange(m_pageHost, m_pageStorage, m_pageNetwork);
 			}
 			else if (type == typeof(ImportSelectHostPage))
 			{
 			    TargetConnection = m_pageHost.ChosenItem == null ? null : m_pageHost.ChosenItem.Connection;
                 RemovePage(m_pageRbac);
                 ConfigureRbacPage(TargetConnection);
+
+				if (m_pageNetwork.Connection != TargetConnection
+					|| m_pageStorage.VmMappings != m_vmMappings

```

Comment:
The m_vmMappings are set further down; at this point they have not changed yet, so the comparison with m_pageStorage.VmMappings is always true. However, I suggest using local variables for comparing the connection and mappings (like with the OvfEnvelope in the previous comment).

Also,  the removal/addition of the boot options page is not needed if the import type or connection have not changed.